### PR TITLE
KAFKA-3347 - Configure java to prefer ipv4

### PR DESCRIPTION
**Pull request by jcustenborder**
Modified KAFKA_JVM_PERFORMANCE_OPTS to include -Djava.net.preferIPv4Stack=true. Added an additional space at the end of the string to be consistent with the other variables.

### CODE REVIEW

**Comment by ijuma** [comment]
You already know this, but it's a bit weird to use `KAFKA_JVM_PERFORMANCE` for this.

**Comment by jcustenborder** [comment]
@ijuma I'm open for suggestions. Here are the best options I think. I didn't want to use `KAFKA_OPTS` because it's most likely to be overridden. Should I add a new variable? If so do you have a suggestion of what I should use? 

```
KAFKA_JVM_PERFORMANCE_OPTS
KAFKA_OPTS
```

**Comment by jcustenborder** [comment]
I could create a new KAFKA_NETWORK_OPTS config setting and add it to the script. This could be useful in the case you want to change [proxy settings](https://docs.oracle.com/javase/8/docs/technotes/guides/net/proxies.html) for connect. Let me know whatever is preferred and I'll update the pull request.

**Comment by ijuma** [comment]
I think KAFKA_NETWORK_OPTS would make sense, but let's wait for a committer to weigh in.

**Comment by jcustenborder** [comment]
`KAFKA_NETWORK_OPTS` could be useful for connect. Then we could have people use things like `export KAFKA_NETWORK_OPTS='-Djava.net.preferIPv4Stack=true -Dhttp.proxyHost=1.2.3.4 -Dhttp.proxyPort=8080'` to wire up proxy settings for connectors.

**Comment by ewencp** [comment]
I think the setting is a good default and I think the addition of `KAFKA_NETWORK_OPTS` is a good idea. How do we feel about this being public interface? Hate to complicate things, but as part of CLI technically this is public and may need to go through a KIP. Otherwise I'd say LGTM and commit.

File: bin/kafka-run-class.sh [hunk 1]
```diff
@@ -161,9 +161,15 @@ fi
 
 # JVM performance options
 if [ -z "$KAFKA_JVM_PERFORMANCE_OPTS" ]; then
-  KAFKA_JVM_PERFORMANCE_OPTS="-server -XX:+UseG1GC -XX:MaxGCPauseMillis=20 -XX:InitiatingHeapOccupancyPercent=35 -XX:+DisableExplicitGC -Djava.awt.headless=true"
+  KAFKA_JVM_PERFORMANCE_OPTS="-server -XX:+UseG1GC -XX:MaxGCPauseMillis=20 -XX:InitiatingHeapOccupancyPercent=35 -XX:+DisableExplicitGC -Djava.awt.headless=true "
```
Comment (ijuma) [inline bin/kafka-run-class.sh:164]: Why do we need the space at the end of this and `KAFKA_NETWORK_OPTS` a few lines below?

**Comment by ijuma** [comment]
I think we didn't do KIPs for small changes like this one in the past (similarly we have added options to CLI tools without a KIP).

**Comment by ijuma** [comment]
On second thought, changing the default could affect existing users so it may be worth a mailing list discussion.

**Pull request closed without merging**